File: src/main/java/com/mbayou/kick4k/events/type/ModerationBannedEvent.java
package com.mbayou.kick4k.events.type;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class ModerationBannedEvent extends KickEvent {
    private final EventUser broadcaster;
    private final EventUser moderator;
    private final EventUser bannedUser;
    private final BanMetadata metadata;

    @JsonCreator
    public ModerationBannedEvent(@JsonProperty("broadcaster") EventUser broadcaster,
                                 @JsonProperty("moderator") EventUser moderator,
                                 @JsonProperty("banned_user") EventUser bannedUser,
                                 @JsonProperty("metadata") BanMetadata metadata) {
        this.broadcaster = broadcaster;
        this.moderator = moderator;
        this.bannedUser = bannedUser;
        this.metadata = metadata;
    }

    public EventUser getBroadcaster() {
        return broadcaster;
    }

    public EventUser getModerator() {
        return moderator;
    }

    public EventUser getBannedUser() {
        return bannedUser;
    }

    public BanMetadata getMetadata() {
        return metadata;
    }

    public static String getEventType() {
        return "moderation.banned";
    }

    public static String getEventVersion() {
        return "1";
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/type/ModerationBannedEvent.kt
package com.mbayou.kick4k.events.type

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class ModerationBannedEvent @JsonCreator constructor(
    @JsonProperty("broadcaster") val broadcaster: EventUser?,
    @JsonProperty("moderator") val moderator: EventUser?,
    @JsonProperty("banned_user") val bannedUser: EventUser?,
    @JsonProperty("metadata") val metadata: BanMetadata?,
) : KickEvent() {
    companion object {
        @JvmStatic
        fun getEventType(): String = "moderation.banned"

        @JvmStatic
        fun getEventVersion(): String = "1"
    }
}
